GuessTheNumber.java

```diff
@@ -20,5 +21,187 @@
  * "There are X players with better scores") or if he's at the top of the table.
  * Before the next game, the top 3 scores (which are greater than 0) will be printed
  */
+import java.util.Random;
+import java.util.Scanner;
+
 public class GuessTheNumber {
+	public static void main(String[] args) {
+		int numberOfSteps = 0, userId, theNumber, theNumberLength, userNumber = 0;
+		int[] highScoreTable = new int[100];
+		int[] highScoreTableSorted = new int[100];
+		String[] messArr = new String[100];
+
+		int choise = 0;
+		Scanner in = new Scanner(System.in);
+
+		System.out.println("\n---- Guess The Number Game -----\n");
+
+		do {
+			numberOfSteps = 0;
+			theNumber = 0;
+
+			// Input from user the player ID.
+			do {
+				System.out.print("Enter Your Player ID (from 1 to 100):");
+				userId = in.nextInt();
+			} while (userId < 1 || userId > 100);
+
+			// Input from player length of the number to guess.
+			do {
+				System.out
+						.print("Enter the number of digits to guess (the difficulty level of the game), from 2 to 9:");
+				theNumberLength = in.nextInt();
+			} while (theNumberLength < 2 || theNumberLength > 9);
+
+			// Generate the Number by the player selected value.
+			for (int i = 0; i < theNumberLength; i++) {
+				Random rn = new Random();
+				int randomNum;
+				if (i == 0) {
+					// First digit in the number can't be 0
+					randomNum = rn.nextInt(9) + 1;
+				} else {
+
+					// Number must be with only unique digits.
+					boolean flag = false;
+					do {
+						flag = false;
+						randomNum = rn.nextInt(10);
+
+						// Check if the generated random digit already exist in the prev. digits in the
+						// number, if yes - exit and generate new random digit.
+						int tmpNum = 0, tmptheNumber = theNumber;
+						for (int j = (theNumberLength - 1); (j > (theNumberLength - 1 - i) && flag == false); j--) {

```

Comment:
'flag == false' can be converted to '!flag'. 
if you change the name to 'duplicateDigitsFound' then it would even make more sense to write it with '!' operator

Reply:
Yes, you are right, fixed..